### ISSUE TITLE
Is latency measurement always ON ?

### ISSUE THREAD

**Issue by ashutshkumr** [open-traffic-generator/models#17]
Should API server always provide latency measurement by default ? If not, how does a user specify that they want to turn off latency measurements ?

### Use Case
Latency measurement demands frame sizes to be more than 64B in certain cases (when Ethernet/IP/TCP is configured, minimum size increases to 80B) - from user's perspective, they'll just get a warning that frame size has been adjusted.

What if user wants to exercise 64B frame size and doesn't really care about latency measurements ?

**Comment by ajbalogh** [comment]
There is a port options model for location_preemption that allows a user to forcibly take a port.
A flow options model should have a property for disabling/enabling latency measurements.
I will add that in a separate pull request for you to review.

**Comment by ashutshkumr** [comment]
Shouldn't configuration for "disabling/enabling latency measurements"  live next to configuration for "custom field tracking" and "latency binning" ? (both of which currently don't exist)

**Comment by ashutshkumr** [comment]
@ajbalogh @ankur-sheth @winstonliu1111, Matt had raised concerns about opting out of ixia instrumentation (internally turning on/off flow stats). As an afterthought, the scope of these settings should be `Config` (since they should apply to all ports or none).

Quick proposal:

```python
c = Config(
    # we already have options field for misc / global settings
    options=Options(
        stat_options=StatOptions(
            # boolean for toggling flow stats
            # if disabled, no instrumentation details will be embedded in egress frames
            enable_flow_stats=True,
            # an ENUM with values 'disabled', 'fifo', 'lifo', 'filo', 'lilo'
            # if not disabled, will provide latency measurements,
            # but could increase the frame size from what user specified (read previous comments for more info).
            latency_measurement='disabled'
        )
    )
)
```

We can later expand `StatOptions` to include configs for latency bins, sequence checking, DI, L1 rate, etc.

**Comment by ankur-sheth** [comment]
I'm still not convinced that we want to or need to expose this to the user.  But there is a need to clearly indicate to the user that payload is over-written for the purposes of flow tracking and latency measurements.  This needs to be documented and also returned as a warning to the user.  
@winstonliu1111 what do you think?  Should we expose these options right now?

**Comment by ajbalogh** [comment]
I agree that it should just be a warning and details can be exposed in the /results/capabilties.

**Comment by winstonliu1111** [comment]
Leaving it as warnings give us some time to delay this decision until we get more asks and feedback. 

Btw, at some point we should maybe exercise the capabilities endpoint and explore what it really means to an end user in a script.  I'm having trouble seeing how that code will work nicely.

**Comment by ashutshkumr** [comment]
**TL;DR**: Fair number of users (and their TCs) won't need flow stats, so they might want to get rid of unwanted packet modifications.


I am yet to see something similar to flow stats being used in SONiC scripts at least (and few other ptf-based scripts I searched for in github gists).

The **USP of flow stats** are latency / jitter / sequence checking / DI metrics, with much higher accuracy, especially on HW Ixia Ports.

So there is a very good reason for users to exercise flow stats, given they're aware of its advantages over **other means [1]** and that it requires packet contents to be modified.

But, if users have no need for such stats, they might as well get rid of unexplained packet modifications.

Another point is (and I need to research more to provide concrete example here), there are scenarios where **DUTs can append headers** to packets being forwarded, overwriting instrumentation signature thusly. 

Lastly, if not for disabling flow stats, we'll ultimately need to expose configuration options related to various flow stats metrics.
(e.g. measure jitter instead of latency, measure sequence checking with certain threshold, etc.)

**[1]** Snippet for less accurate latency measurement borrowed from Scapy issue 1791:
```python
dest = "10.0.0.1"

pkt = []
for i in range(1,50):
    a = IP(dst=dest) / TCP(flags="S", seq=i, sport=65000, dport=55556)
    pkt.append(a)

ans, unans = sr(pkt, filter="host {0}".format(dest), inter=0, timeout=1)

print("scapy version: {}".format(conf.version))

for pkt in ans:
    sent = pkt[0]
    received = pkt[1]
    res = (received.time - sent.sent_time) * 1000
    print(res)
```

**Comment by ajbalogh** [comment]
With version 0.4.0 latency by default is always disabled.  
There is an option to enable latency on a per flow basis under the flow.metrics.latency container.
